docs/design/013-caching/README.md

```diff
@@ -0,0 +1,113 @@
+# Caching of data from ledger
+
+Currently whenever credential definitions and/or schemas is needed, it is being fetched from the ledger.
+This operation may last multiple seconds and is slowing down usage of credentials.
+Caching also enables usage of anoncreds in areas where user do not have internet coverage (eg. Using passport credential on foreign airport).
+
+## Goals and ideas
+
+* Allow users to cache credential definitions and schemas.
+  * Local wallet to be used because although this data is public, possession of some credential definition or schema reveals possession of respective credential. 
+* Provide higher level api for fetching this data so it is easier to use.
+  * Caching should be transparent to the user.
+* Enable purging of old (not needed more) data.
+  
+## Public API
+
+Note: In all calls `pool_handle` may be removed if did resolver is implemented.
+
+```Rust
+/// Gets credential definition json data for specified credential definition id.
+/// If data is present inside of cache, cached data is returned.
+/// Otherwise data is fetched from the ledger and stored inside of cache for future use.
+/// 
+/// #Params
+/// command_handle: command handle to map callback to caller context.
+/// pool_handle: pool handle (created by open_pool_ledger).
+/// wallet_handle: wallet handle (created by open_wallet).
+/// submitter_did: DID of the submitter stored in secured Wallet.
+/// id: identifier of credential definition.
+/// options_json:
+///  {
+///    forceUpdate: (optional, false by default) Force update of record in cache from the ledger,
```

Comment:
I believe we can use  something similar to http-cache control headers:

```http
Cache-Control: max-age=<seconds>
Cache-Control: max-stale[=<seconds>]
Cache-Control: min-fresh=<seconds>
Cache-Control: no-cache 
Cache-Control: no-store
Cache-Control: no-transform
Cache-Control: only-if-cached
```

Reply:
Some of these options look very useful. I would like not to have too many options, which may be too confusing, but like some, max-age, only-if-cached, for eg.